codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2PushPromiseFrameTest.java

```diff
@@ -0,0 +1,235 @@
+
+        private ChannelHandlerContext ctx;
+
+        @Override
+        public void channelActive(ChannelHandlerContext ctx) throws InterruptedException {
+            this.ctx = ctx;
+        }
+
+        void write() throws InterruptedException {
+            Http2Headers http2Headers = new DefaultHttp2Headers();
+            http2Headers.path("/")
+                    .authority("localhost")
+                    .method("GET")
+                    .scheme("https");
+
+            Http2HeadersFrame headersFrame = new DefaultHttp2HeadersFrame(http2Headers, true);
+            headersFrame.stream(newStream());
+            ctx.writeAndFlush(headersFrame);
+        }
+
+        @Override
+        public void channelRead(ChannelHandlerContext ctx, Object msg) {
+
+            if (msg instanceof Http2PushPromiseFrame) {
+                Http2PushPromiseFrame pushPromiseFrame = (Http2PushPromiseFrame) msg;
+
+                assertEquals("/meow", pushPromiseFrame.http2Headers().path().toString());
+                assertEquals("GET", pushPromiseFrame.http2Headers().method().toString());
+                assertEquals("https", pushPromiseFrame.http2Headers().scheme().toString());
+                assertEquals("localhost:5555", pushPromiseFrame.http2Headers().authority().toString());
+
+                Http2PriorityFrame priorityFrame = new DefaultHttp2PriorityFrame(pushPromiseFrame.stream().id(),
+                        Http2CodecUtil.DEFAULT_PRIORITY_WEIGHT, true);
+                priorityFrame.stream(pushPromiseFrame.pushStream());
+                ctx.writeAndFlush(priorityFrame);
+            } else if (msg instanceof Http2HeadersFrame) {
+                Http2HeadersFrame headersFrame = (Http2HeadersFrame) msg;
+
+                if (headersFrame.stream().id() == 3) {
+                    assertEquals("200", headersFrame.headers().status().toString());
+                    assertEquals("false", headersFrame.headers().get("push").toString());
+                } else if (headersFrame.stream().id() == 2) {
+                    assertEquals("200", headersFrame.headers().status().toString());
+                    assertEquals("true", headersFrame.headers().get("push").toString());
+                } else {
+                    ctx.writeAndFlush(new DefaultHttp2GoAwayFrame(Http2Error.REFUSED_STREAM));
+                }
+            } else if (msg instanceof Http2DataFrame) {
+                Http2DataFrame dataFrame = (Http2DataFrame) msg;
+
+                if (dataFrame.stream().id() == 3) {
+                    assertEquals("Meow", dataFrame.content().toString(CharsetUtil.UTF_8));
+                    dataFrame.content().release();
+                } else if (dataFrame.stream().id() == 2) {
+                    assertEquals("Meow, I am Pushed via HTTP/2", dataFrame.content().toString(CharsetUtil.UTF_8));
+                    dataFrame.content().release();
+                } else {
+                    dataFrame.content().release();
```

Comment:
consider removing all these release and just add a big `try {... } finally { ReferenceCountUtil.release(msg); }`